pkg/apis/kubermatic/v1/configuration.go

```diff
@@ -205,6 +205,14 @@ type KubermaticSeedControllerConfiguration struct {
 	DebugLog bool `json:"debugLog,omitempty"`
 	// Replicas sets the number of pod replicas for the seed-controller-manager.
 	Replicas *int32 `json:"replicas,omitempty"`
+	// AddonCollector enables addon metrics collector, Default value is `true`.
+	AddonCollector *bool `json:"addonCollector,omitempty"`
+	// ClusterBackupCollector enables backup cluster metrics collector, Default value is `true`.
+	ClusterBackupCollector *bool `json:"clusterBackupCollector,omitempty"`
+	// ClusterCollector enables cluster metrics collector, Default value is `true`.
+	ClusterCollector *bool `json:"clusterCollector,omitempty"`
+	// ProjectCollector enables project metrics collector, Default value is `true`.
+	ProjectCollector *bool `json:"projectCollector,omitempty"`
```

Comment:
Couldn't this be a `EnabledCollectors []string` ? Having one field per collector seems excessive.

Reply:
Agreed. This seems like a better idea, you can make sure that the empty list is defaulted to include all existing collectors by adding defaulting in [DefaultConfiguration](https://github.com/kubermatic/kubermatic/blob/d6685584e4e6f866e4e32aadd3cc2b2feb04ff0a/pkg/defaulting/configuration.go#L369).

Reply:
I added `DisabledCollectors` instead of `EnabledCollectors` because, in this case, an empty list (the default value) means that all collectors are enabled. This approach is backwards compatible and eliminates the need to modify the defaulting logic.